src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/DotNetHost.cs

```diff
@@ -18,13 +18,23 @@ public partial class DotNetHost
         public static Version RuntimeVersion =>
             s_runtimeVersionLazy.Value;
 
-        public static string HostExeNameWithoutExtension => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-            Path.GetFileNameWithoutExtension(HostExePath) :
-            Path.GetFileName(HostExePath);
+        public static string ExeName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "dotnet.exe" : "dotnet";
 
-        public static string HostExePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-            @"..\..\..\..\..\.dotnet\dotnet.exe" :
-            "../../../../../.dotnet/dotnet";
+        public static string ExeNameWithoutExtension => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+            Path.GetFileNameWithoutExtension(ExeName) :
+            ExeName;
+
+        public static string GetPath(Architecture? arch = null)
+        {
+            // e.g. <repoPath>/.dotnet
+            string dotnetDirPath = Path.Combine("..", "..", "..", "..", "..", ".dotnet");
```

Comment:
Are there cases where we don't want to use this host but rather the one on the machine itself? For example, VS unit test scenarios?

Reply:
I'll think about how we can handle that and save it for another PR